File: 06_14.03_front-end/kymnevoistlus/kymnevoistlus/src/models/Athlete.ts
export interface Result {
  id: number;
  event: string;
  result: number;
  points: number;
}

export interface Athlete {
    id: number;
    name: string;
    country: string;
    age: number;
    totalPoints: number;
    results: {
        [event: string]: number;
    };
}

File: 01_07.02_andmemudelid/kumnevoistlus/src/main/java/martintk/kumnevoistlus/entity/Athlete.java
package martintk.kumnevoistlus.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Athlete {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String country;
    private double age;

    @OneToMany(mappedBy = "athlete", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Result> results;
}
